api/types.proto

```diff
@@ -713,3 +713,33 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretType provides information about what kind of secret this is
+enum SecretType {
+	CONTAINER = 0 [(gogoproto.enumvalue_customname) = "ContainerSecret"];
+	NODE = 1 [(gogoproto.enumvalue_customname) = "NodeSecret"];
+}
+
+// SecretReference is the linkage between a service and a secret that it uses.
+message SecretReference {
+	// Name is the name of the secret that this reference.
+	string name = 1;
+
+	// SecretDataID represents the specific Secret version that we're referencing to.
+	// This identifier exists so that SecretReferences don't leak any information
+	// about the secret contents.
+	string secret_data_id = 2 [(gogoproto.customname) = "SecretDataID"];
```

Comment:
Is there any advantage in doing so ? Since we are referring to the same Secret but different versions it seems natural to list the versions but keep one reference.